File: Ruby/response/mms/InboundMessageNotification.rb
require "response/mms/InboundMessage"

class InboundMessageNotification

	def initialize
		@callbackData=nil
		@inboundMessage=nil
	end

	def initializeJSON(jsondict)
		@callbackData=nil
		if (jsondict!=nil) && (jsondict.has_key?'callbackData') && (jsondict['callbackData']!=nil)
			@callbackData=jsondict['callbackData']
		end
		@inboundMessage=nil
		if (jsondict!=nil) && (jsondict.has_key?'inboundMessage') && (jsondict['inboundMessage']!=nil) then
			@inboundMessage=InboundMessage.new
			@inboundMessage.initializeJSON(jsondict['inboundMessage'])
		end
	end

	def getCallbackData
		@callbackData
	end

	def setCallbackData(callbackData)
		@callbackData=callbackData
	end

	def getInboundMessage
		@inboundMessage
	end

	def setInboundMessage(inboundMessage)
		@inboundMessage=inboundMessage
	end


end 

File: Python/response/mms/InboundMessageNotification.py
from response.mms.InboundMessage import InboundMessage

class InboundMessageNotification:
	"""the inboundMessageNotification object includes any callbackData and an inboundMMSMessage array with message detail elements"""
	
	def __init__(self):
		"""Default class constructor"""
		self.callbackData=None
		self.inboundMessage=None
	
	def __init__(self, jsondict):
		"""Class constructor that will create an instance initialised from a parsed JSON data block"""
		self.callbackData=None
		if jsondict is not None and 'callbackData' in jsondict and jsondict['callbackData'] is not None:
			self.callbackData=jsondict['callbackData']
		self.inboundMessage=None
		if jsondict is not None and 'inboundMessage' in jsondict and jsondict['inboundMessage'] is not None:
			self.inboundMessage=InboundMessage(jsondict['inboundMessage'])
	
	def getCallbackData(self):
		"""getter for callbackData : specifies the information that will accompany a notification"""
		return self.callbackData
	
	def setCallbackData(self,callbackData):
		"""setter for callbackData : specifies the information that will accompany a notification"""
		self.callbackData=callbackData
	
	def getInboundMessage(self):
		"""getter for inboundMessage : specifies the message details including sender, recipient"""
		return self.inboundMessage
	
	def setInboundMessage(self,inboundMessage):
		"""setter for inboundMessage : specifies the message details including sender, recipient"""
		self.inboundMessage=inboundMessage
	
